dll/win32/shell32/CDefView.cpp

```diff
@@ -767,40 +833,239 @@ void CDefView::UpdateListColors()
 */
 BOOL CDefView::InitList()
 {
-    SHELLDETAILS sd;
-    WCHAR szTemp[50];
     HIMAGELIST big_icons, small_icons;
 
     TRACE("%p\n", this);
 
     m_ListView.DeleteAllItems();
 
+    Shell_GetImageLists(&big_icons, &small_icons);
+    m_ListView.SetImageList(big_icons, LVSIL_NORMAL);
+    m_ListView.SetImageList(small_icons, LVSIL_SMALL);
+
     m_hMenuArrangeModes = CreateMenu();
+    LoadColumns();
+    return TRUE;
+}
+
+/**********************************************************
+* Column handling
+*/
+static HRESULT SHGetLVColumnSubItem(HWND List, UINT Col)
+{
+    LVCOLUMN lvc;
+    lvc.mask = LVCF_SUBITEM;
+    if (!ListView_GetColumn(List, Col, &lvc))
+        return E_FAIL;
+    else
+        return lvc.iSubItem;
+}
+
+HRESULT CDefView::MapFolderColumnToListColumn(UINT FoldCol)
+{
+    // This function is only called during column management, performance is not critical.
+    for (UINT i = 0;; ++i)
+    {
+        HRESULT r = SHGetLVColumnSubItem(m_ListView.m_hWnd, i);
+        if ((UINT) r == FoldCol)
+            return i;
+        else if (FAILED(r))
```

Comment:
```suggestion
        else if (r == E_FAIL)
```

Reply:
It might return another error in the future, who knows. That is why those macros exit.